lib/membrane/rtp/twcc_sender/receiver_rate.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Membrane.RTP.TWCCSender.ReceiverRate do
+  @moduledoc false
+  # module responsible for calculating bitrate
+  # received by a receiver in last `window` time
+
+  alias Membrane.Time
+
+  @type t() :: %__MODULE__{
+          value: float() | nil,
+          # time window for measuring the received bitrate, between [0.5, 1]s (reffered to as "T" in the RFC)
+          window: Time.t(),
+          # accumulator for packets and their timestamps that have been received through target_receive_interval
+          packets_received: [{Time.t(), pos_integer()}]
```

Comment:
The `List.last(packets_received)` in second `update` clause will be faster but on the other hand we had to switch from `Enum.drop_while` to `:queue.filter` and the `:queue` API is pretty wierd. Take a look here https://github.com/membraneframework/membrane_rtp_plugin/pull/115/commits/51cee7277bf0ed98725cd5587ef21e9a6615f4db

Reply:
How about `Qex` then? It's already a `membrane_core` dependency and implements Elixir protocols, including `Enumerable`

Reply:
Looking at how they implement `Enumerable` https://github.com/princemaple/elixir-queue/blob/v0.5.1/lib/impls/enumerable.ex I think we will get similar results to the list implementation as every time we want to iterate over the queue we are converting it to a list at first and converting to a list is O(n) https://github.com/erlang/otp/blob/master/lib/stdlib/src/queue.erl#L89

So to sum up
1. current implementation should be the fastest one
2. Qex and list implementations should have similar performance: (taking last element, drop_while)
O(1) + O(n) +O(x) - Qex
O(n) + O(x) - list

Reply:
Well, both `:queue` and `Qex` could use a recursive function implementing the `drop_while` without step O(n).

> current implementation should be the fastest one

I wouldn't be so sure as `filter` is also O(n), so it would all depend on whether O(x) is closer to O(1) or O(n) - and by this code, I was assuming the former.
Anyway, I wouldn't be surprised if all of our estimations were wrong, as most of the complexities for `:queue` are amortized and might degrade due to the way the calls are ordered, not to mention that for only 25 elements the big O complexity might not be that relevant.

So from my perspective, unless benchmarks tip the scale in favour of one of the options, I'd choose the most readable one. I felt that `packets_received` was a FIFO, and API-wise `Qex` > `:queue`, so this is what I would choose, but the difference seems marginal now and I'll accept any option. :wink: